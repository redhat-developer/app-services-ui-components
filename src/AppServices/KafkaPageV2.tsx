import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderMain,
  CardTitle,
  DescriptionList,
  DescriptionListDescription,
  DescriptionListGroup,
  DescriptionListTerm,
  Divider,
  Flex,
  FlexItem,
  Grid,
  Split,
  SplitItem,
  Stack,
  Text,
  TextContent,
  TextVariants,
  Title,
} from "@patternfly/react-core";
import type { FunctionComponent } from "react";
import { Trans, useTranslation } from "react-i18next";
import { AppSpeedKeyArt, LogoAWS, LogoAzure } from "../images";
import {
  KafkaInstanceCapacityTable,
  MarketingPageHero,
  MarketingPageSection,
} from "./components";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";

export const KafkaPageV2: FunctionComponent = () => {
  const { t } = useTranslation(["kafkaoverview-v2"]);
  return (
    <>
      <MarketingPageHero
        title={t("heroTitle")}
        tagLine={t("heroTagline")}
        description={t("heroDescription")}
        heroImage={AppSpeedKeyArt}
        heroImageSize={478}
        heroImageCanRepeat={false}
        heroImagePositionY={-99}
      />
      <MarketingPageSection>
        <Grid hasGutter lg={6}>
          <Card
            ouiaId="card-overview-purchase-now"
            aria-label={t("kafkaoverview:purchaseCardTitle")}
          >
            <CardHeader>
              <CardHeaderMain>
                <CardTitle>
                  <Title headingLevel="h3">{t("purchaseCardTitle")}</Title>
                </CardTitle>
              </CardHeaderMain>
            </CardHeader>
            <CardBody>{t("purchaseCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardPurchase-buttonCTA"
                variant={ButtonVariant.link}
                isLarge
                isInline
                component="a"
                href="https://marketplace.redhat.com/en-us/products/red-hat-openshift-streams-for-apache-kafka"
                target="_blank"
                ouiaId="link-purchase-now"
              >
                {t("purchaseCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
          <Card
            ouiaId="card-overview-contact-sales"
            aria-label={t("kafkaoverview:contactSalesCardTitle")}
          >
            <CardHeader>
              <CardTitle>
                <Title headingLevel="h2">{t("contactSalesCardTitle")}</Title>
              </CardTitle>
            </CardHeader>
            <CardBody>{t("contactSalesCardMainText")}</CardBody>
            <CardFooter>
              <Button
                data-testid="cardContactSales-buttonCTA"
                variant={ButtonVariant.link}
                isLarge
                isInline
                component="a"
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka#contact-us"
                target="_blank"
                ouiaId="link-contact-sales"
              >
                {t("contactSalesCardCallToActionButton")}
                <ExternalLinkAltIcon className="pf-u-ml-md" />
              </Button>
            </CardFooter>
          </Card>
        </Grid>
      </MarketingPageSection>
      <MarketingPageSection
        className={"kafka-overview--page-section--marketing"}
      >
        <Card>
          <CardTitle>
            <Title headingLevel="h2">{t("pricingModalTitle")}</Title>
          </CardTitle>
          <CardBody>
            <Flex spaceItems={{ default: "spaceItemsXl" }}>
              <Flex
                alignSelf={{ default: "alignSelfCenter" }}
                justifyContent={{ default: "justifyContentCenter" }}
                flex={{ default: "flex_1" }}
              >
                <FlexItem>
                  <DescriptionList isHorizontal>
                    <DescriptionListGroup>
                      <DescriptionListTerm>
                        {t("steamingUnit")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("streamingUnit_price")}
                      </DescriptionListDescription>
                    </DescriptionListGroup>
                    <DescriptionListGroup>
                      <DescriptionListTerm>
                        {t("dataTransfer")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("dataTransfer_price")}
                      </DescriptionListDescription>
                    </DescriptionListGroup>
                    <DescriptionListGroup>
                      <DescriptionListTerm>{t("storage")}</DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("storage_price")}
                      </DescriptionListDescription>
                    </DescriptionListGroup>
                  </DescriptionList>
                </FlexItem>
              </Flex>
              <Divider isVertical />
              <FlexItem flex={{ default: "flex_1" }}>
                <TextContent>
                  <Text component={TextVariants.p}>
                    <Trans
                      ns={"kafkaoverview-v2"}
                      i18nKey={"streamingUnitText"}
                      components={{ bold: <strong /> }}
                    />
                  </Text>
                </TextContent>
              </FlexItem>
            </Flex>
          </CardBody>
        </Card>
      </MarketingPageSection>
      <MarketingPageSection
        className={"kafka-overview--page-section--marketing"}
      >
        <Card>
          <CardTitle>
            <Title headingLevel="h2">{t("kafkaInstanceCapacityTitle")}</Title>
          </CardTitle>
          <CardBody>
            <Stack hasGutter>
              <Text component={TextVariants.p}>
                {t("kafkaInstanceCapacityDescription")}
              </Text>
              <KafkaInstanceCapacityTable />
            </Stack>
          </CardBody>
          <CardFooter>
            <TextContent className={"pf-u-font-size-sm"}>
              <Trans
                ns={"kafkaoverview-v2"}
                i18nKey={"kafkaInstanceCapacityfooter"}
                components={[<SupportLink />, <MonitoringDiskSpaceLink />]}
              />
            </TextContent>
          </CardFooter>
        </Card>
      </MarketingPageSection>
      <MarketingPageSection
        className={"kafka-overview--page-section--marketing"}
      >
        <Card>
          <CardTitle>
            <Title headingLevel="h2">{t("cloudProvidersTitle")}</Title>
          </CardTitle>
          <CardBody>
            <Flex spaceItems={{ default: "spaceItemsXl" }}>
              <Flex
                justifyContent={{ default: "justifyContentCenter" }}
                flex={{ default: "flex_1" }}
              >
                <FlexItem>
                  <Split hasGutter>
                    <SplitItem>
                      <img src={LogoAWS} alt={""} style={{ height: "60px" }} />
                    </SplitItem>
                    <SplitItem>
                      <Title headingLevel="h2" className={"pf-u-pt-sm"}>
                        {t("awsWebServiceTitle")}
                      </Title>
                    </SplitItem>
                  </Split>
                </FlexItem>
              </Flex>
              <Divider isVertical />
              <Flex
                justifyContent={{ default: "justifyContentCenter" }}
                flex={{ default: "flex_1" }}
              >
                <FlexItem>
                  <Split hasGutter>
                    <SplitItem>
                      <img
                        src={LogoAzure}
                        alt={""}
                        style={{ height: "60px" }}
                      />
                    </SplitItem>
                    <SplitItem>
                      <Title headingLevel="h2" className={"pf-u-pt-sm"}>
                        {t("microsoftAzureTitle")}
                      </Title>
                      <Text
                        component={TextVariants.p}
                        className={"pf-u-color-200"}
                      >
                        {t("microsoftAzureDescription")}
                      </Text>
                    </SplitItem>
                  </Split>
                </FlexItem>
              </Flex>
            </Flex>
          </CardBody>
        </Card>
      </MarketingPageSection>
    </>
  );
};

const SupportLink: FunctionComponent = ({ children }) => (
  <Button
    data-testid="redhatSupport-buttonCTA"
    isInline
    variant={ButtonVariant.link}
    component="a"
    target="_blank"
    href="https://access.redhat.com/support"
  >
    {children}
    <ExternalLinkAltIcon className="pf-u-ml-xs" />
  </Button>
);

const MonitoringDiskSpaceLink: FunctionComponent = ({ children }) => (
  <Button
    data-testid="diskSpaceMetrics-buttonCTA"
    isInline
    variant={ButtonVariant.link}
    component="a"
    target="_blank"
    href="https://access.redhat.com/documentation/en-us/red_hat_openshift_streams_for_apache_kafka/1/guide/aced8e5e-8229-4cb2-82f9-87a8caa24bb3"
  >
    {children}
    <ExternalLinkAltIcon className="pf-u-ml-xs" />
  </Button>
);
