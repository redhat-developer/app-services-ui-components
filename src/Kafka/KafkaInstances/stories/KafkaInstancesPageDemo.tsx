import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
