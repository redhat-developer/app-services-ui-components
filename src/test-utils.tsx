import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
