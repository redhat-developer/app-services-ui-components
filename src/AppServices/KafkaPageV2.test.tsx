import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../test-utils";
import * as stories from "./KafkaPageV2.stories";

const { KafkaPageV2 } = composeStories(stories);

describe("KafkaPage", () => {
  it("renders", async () => {
    const comp = render(<KafkaPageV2 />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });

  it("should persist ouia ids", async () => {
    const comp = render(<KafkaPageV2 />);
    await waitForI18n(comp);
    expect(comp.getByRole("article", { name: "Purchase now" })).toHaveAttribute(
      "data-ouia-component-id",
      "card-overview-purchase-now"
    );
    expect(comp.getByRole("link", { name: "Purchase now" })).toHaveAttribute(
      "data-ouia-component-id",
      "link-purchase-now"
    );

    expect(
      comp.getByRole("article", { name: "Contact sales" })
    ).toHaveAttribute("data-ouia-component-id", "card-overview-contact-sales");
    expect(comp.getByRole("link", { name: "Contact sales" })).toHaveAttribute(
      "data-ouia-component-id",
      "link-contact-sales"
    );
  });
});
