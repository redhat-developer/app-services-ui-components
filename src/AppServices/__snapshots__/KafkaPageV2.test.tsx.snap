// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaPage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-light appsrv-marketing--banner pf-u-background-color-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 478px; --appsrv-marketing--banner--before--BackgroundRepeat: no-repeat; --appsrv-marketing--banner--before--BackgroundPositionY: -99px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift Streams for Apache Kafka
        </h1>
        <p
          class="appsrv-marketing--banner__tagline pf-u-color-200"
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Fully hosted and managed service for stream-based applications
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Red Hat OpenShift Streams for Apache Kafka is a managed cloud service that provides a streamlined developer experience for building, deploying, and scaling real-time applications in hybrid-cloud environments. The combination of seamless operations across distributed microservices, large data transfer volumes, and managed operations allows teams to focus on core competencies, accelerate application velocity and reduce operational cost.
        </p>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-all-6-col-on-lg pf-m-gutter"
        >
          <article
            aria-label="Purchase now"
            class="pf-c-card"
            data-ouia-component-id="card-overview-purchase-now"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class=""
              >
                <div
                  class="pf-c-card__title"
                >
                  <h3
                    class="pf-c-title pf-m-lg"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Purchase now
                  </h3>
                </div>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              In the United States or Canada, purchase a pay-as-you-go subscription for up to 25 streaming units through Red Hat Marketplace.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline pf-m-display-lg"
                data-ouia-component-id="link-purchase-now"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardPurchase-buttonCTA"
                href="https://marketplace.redhat.com/en-us/products/red-hat-openshift-streams-for-apache-kafka"
                target="_blank"
              >
                Purchase now
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-md"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
          <article
            aria-label="Contact sales"
            class="pf-c-card"
            data-ouia-component-id="card-overview-contact-sales"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__header"
            >
              <div
                class="pf-c-card__title"
              >
                <h2
                  class="pf-c-title pf-m-xl"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                  Contact sales
                </h2>
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              Contact sales to get a pre-paid subscription or modify limits of your current subscription.
            </div>
            <div
              class="pf-c-card__footer"
            >
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline pf-m-display-lg"
                data-ouia-component-id="link-contact-sales"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="cardContactSales-buttonCTA"
                href="https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka#contact-us"
                target="_blank"
              >
                Contact sales
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-md"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </article>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width kafka-overview--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <article
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h2
              class="pf-c-title pf-m-xl"
              data-ouia-component-id="OUIA-Generated-Title-4"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Pricing model
            </h2>
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-space-items-xl"
            >
              <div
                class="pf-l-flex pf-m-flex-1 pf-m-align-self-center pf-m-justify-content-center"
              >
                <div
                  class=""
                >
                  <dl
                    class="pf-c-description-list pf-m-horizontal"
                  >
                    <div
                      class="pf-c-description-list__group"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Streaming unit
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          $1.49 / hour
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-c-description-list__group"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Data transfer
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          $0.09 / GB
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-c-description-list__group"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Storage
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          $0.10 / GB month
                        </div>
                      </dd>
                    </div>
                  </dl>
                </div>
              </div>
              <hr
                class="pf-c-divider pf-m-vertical"
              />
              <div
                class="pf-m-flex-1"
              >
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-3"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    A 
                    <strong>
                      streaming unit
                    </strong>
                     determines the default maximum capacity of a Kafka instance. The more streaming units your Kafka instance has, the larger its capacity will be.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </article>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width kafka-overview--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <article
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-2"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h2
              class="pf-c-title pf-m-xl"
              data-ouia-component-id="OUIA-Generated-Title-5"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Kafka instance capacity
            </h2>
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-stack pf-m-gutter"
            >
              <p
                class=""
                data-ouia-component-id="OUIA-Generated-Text-4"
                data-ouia-component-type="PF4/Text"
                data-ouia-safe="true"
                data-pf-content="true"
              >
                Streams for Apache Kafka offers instances in the following sizes with the following capacity.
              </p>
              <table
                class="pf-c-table pf-m-grid-md"
                data-ouia-component-id="OUIA-Generated-Table-1"
                data-ouia-component-type="PF4/Table"
                data-ouia-safe="true"
                role="grid"
              >
                <thead
                  class=""
                >
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-1"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <th
                      class=""
                    />
                    <th
                      class=""
                      scope="col"
                    >
                      1 streaming unit
                    </th>
                    <th
                      class=""
                      scope="col"
                    >
                      2 streaming units
                    </th>
                  </tr>
                </thead>
                <tbody
                  class=""
                  role="rowgroup"
                >
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-2"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Ingress (MB/second)
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 50
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 100
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-3"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Egress (MB/second)
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 100
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 200
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-4"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Storage* (GB)
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 1000
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 2000
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-5"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Topic partitions
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 1500
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 3000
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-6"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Client connections
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 3000
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 6000
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-7"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Connection rate (connections/second)
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 100
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 200
                    </td>
                  </tr>
                  <tr
                    class=""
                    data-ouia-component-id="OUIA-Generated-TableRow-8"
                    data-ouia-component-type="PF4/TableRow"
                    data-ouia-safe="true"
                  >
                    <td
                      class=""
                      data-label=""
                    >
                      Message size (MB)
                    </td>
                    <td
                      class=""
                      data-label="1 streaming unit"
                    >
                      up to 1
                    </td>
                    <td
                      class=""
                      data-label="2 streaming units"
                    >
                      up to 1
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="pf-c-card__footer"
          >
            <div
              class="pf-c-content pf-u-font-size-sm"
            >
              *Storage can be extended. If you wish to increase the storage limit for a Kafka instance, contact 
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="redhatSupport-buttonCTA"
                href="https://access.redhat.com/support"
                target="_blank"
              >
                Red Hat support
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-xs"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
               to discuss your options. You can check your current storage usage by 
              <a
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="OUIA-Generated-Button-link-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="diskSpaceMetrics-buttonCTA"
                href="https://access.redhat.com/documentation/en-us/red_hat_openshift_streams_for_apache_kafka/1/guide/aced8e5e-8229-4cb2-82f9-87a8caa24bb3"
                target="_blank"
              >
                monitoring disk space metrics
                <svg
                  aria-hidden="true"
                  class="pf-u-ml-xs"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
              .
            </div>
          </div>
        </article>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width kafka-overview--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <article
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-3"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__title"
          >
            <h2
              class="pf-c-title pf-m-xl"
              data-ouia-component-id="OUIA-Generated-Title-6"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Cloud providers
            </h2>
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-space-items-xl"
            >
              <div
                class="pf-l-flex pf-m-flex-1 pf-m-justify-content-center"
              >
                <div
                  class=""
                >
                  <div
                    class="pf-l-split pf-m-gutter"
                  >
                    <div
                      class="pf-l-split__item"
                    >
                      <img
                        alt=""
                        style="height: 60px;"
                      />
                    </div>
                    <div
                      class="pf-l-split__item"
                    >
                      <h2
                        class="pf-c-title pf-m-xl pf-u-pt-sm"
                        data-ouia-component-id="OUIA-Generated-Title-7"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe="true"
                      >
                        Amazon Web Services
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
              <hr
                class="pf-c-divider pf-m-vertical"
              />
              <div
                class="pf-l-flex pf-m-flex-1 pf-m-justify-content-center"
              >
                <div
                  class=""
                >
                  <div
                    class="pf-l-split pf-m-gutter"
                  >
                    <div
                      class="pf-l-split__item"
                    >
                      <img
                        alt=""
                        style="height: 60px;"
                      />
                    </div>
                    <div
                      class="pf-l-split__item"
                    >
                      <h2
                        class="pf-c-title pf-m-xl pf-u-pt-sm"
                        data-ouia-component-id="OUIA-Generated-Title-8"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe="true"
                      >
                        Microsoft Azure
                      </h2>
                      <p
                        class="pf-u-color-200"
                        data-ouia-component-id="OUIA-Generated-Text-5"
                        data-ouia-component-type="PF4/Text"
                        data-ouia-safe="true"
                        data-pf-content="true"
                      >
                        Coming soon
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
